app/controllers/api_controller.rb

```diff
@@ -64,10 +64,16 @@ def render_bad_request_error(exception)
   end
 
   def render_resource_not_found_error(exception)
+    # NB: cleaner error message without a long WHERE id=foo clause
+    detail = if exception.is_a?(ActiveRecord::RecordNotFound) && exception.id.present?
```

Comment:
is it not redundant to check if 
`exception.is_a?(ActiveRecord::RecordNotFound`
since we have
`rescue_from ActiveRecord::RecordNotFound, with: :render_resource_not_found_error` ?

Reply:
yep, you are right, I'll change it 👍

Reply:
Looks like there's one other case where this is called explicitly (and passes an instance of `Exception`) so the check is needed, but perhaps this calling code (from `PeopleController`) could be cleaned up instead, or possibly adapted to follow the standard pattern with populated 'model', `primary_key` and `id` attributes present):

```ruby
render_resource_not_found_error(Exception.new('Image not found'))
```

Better to change that to just throw an `ActiveRecord::RecordNotFound` error I think, and then let this method catch it - which would then allow the `detail` branching code here to be removed and replaced  with a consistent error message i.e.

```ruby
render(
  json: { errors: [{
    title: 'Resource not found',
    detail: "Couldn't find #{exception.model} with '#{exception.primary_key}'=#{exception.id}",
  }] },
  status: :not_found,
)
```

Reply:
@smoothcontract good spot re people_controller - I'll update that to throw an ActiveRecord::RecordNotFound as you suggest. In some tests I've found that the exception.id field is not always set, so I think an `if` statement is still required, e.g.

```
 if exception.id.present?
    "Couldn't find #{exception.model} with '#{exception.primary_key}'=#{exception.id}"
 else
    exception.to_s
 end
```